### PR TITLE
[API] List events belonging to a group

### PR DESCRIPTION
**Pull request by Jon1969Edwards**
## Tasks
- Create request specs
- Create an association Group `has_many` Events and Event `belongs_to` Group
- Add a `date` attribute to Event
- Add 2 scopes: past_events and future_events
- Add 2 keys to the Group serializer -  past_events and future_events

## PT
[Link](https://www.pivotaltracker.com/story/show/163551280)

### CODE REVIEW

File: spec/requests/list_events_belonging_to_group_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+describe 'GET /groups/:id/events' do
+  let!(:group) { create(:group) } 
+  let!(:events) { 3.times { create(:event) } }
```
Comment (tochman) [inline spec/requests/list_events_belonging_to_group_spec.rb]: These events need to belong to the group you created above. You should also create a few events belonging to another group

Comment (tochman) [reply spec/requests/list_events_belonging_to_group_spec.rb]:
```
 let!(:events) { 3.times { create(:event, group: group) } }
```

Comment (CamronLDNF) [reply spec/requests/list_events_belonging_to_group_spec.rb]: Done

File: config/routes.rb [hunk 2]
```diff
@@ -1,14 +1,17 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
+  get 'groups/index'
+  get 'groups/show'
```
Comment (faraznaeem) [inline config/routes.rb]: Why are you using non-resourceful routes?
please remove the autogenerated routes if they do not have any use.

File: spec/factories/events.rb [hunk 3]
```diff
@@ -1,5 +1,7 @@
 FactoryBot.define do
   factory :event do
-    title { "MyString" }
+    title { "EventName" }
+    date { "12/12/2019" }
```
Comment (faraznaeem) [inline spec/factories/events.rb]: is this the best format for date?

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 2 above)
Comment (CamronLDNF) [reply config/routes.rb]: Noted. Done. Thanks.

File: spec/factories/events.rb (hunk 3 above)
Comment (CamronLDNF) [reply spec/factories/events.rb]: Ah, no, I've updated it to `2019-12-12` format.

**Pull request merged**